src/Razor/Microsoft.AspNetCore.Razor.Language/src/DefaultRazorIntermediateNodeLoweringPhase.cs

```diff
@@ -1214,6 +1213,24 @@ public override void VisitMarkupElement(MarkupElementSyntax node)
                 base.VisitMarkupElement(node);
 
                 _builder.Pop();
+
+                static bool LooksLikeAComponentName(DocumentIntermediateNode document, string startTagName)
+                {
+                    var category = char.GetUnicodeCategory(startTagName, 0);
+
+                    // A markup element which starts with an uppercase character is likely a component.
+                    //
+                    // In certain cultures, characters are not explicitly Uppercase/Lowercase, hence we must check
+                    // the specific UnicodeCategory to see if we may still be able to treat it as a component.
+                    //
+                    //    The goal here is to avoid clashing with any future standard-HTML elements.
+                    //
+                    //    To avoid a breaking change, the support of localized component names (without explicit
+                    //    Uppercase classification) is behind a `SupportLocalizedComponentNames` feature flag.
```

Comment:
```suggestion
                    //  The goal here is to avoid clashing with any future standard-HTML elements.
                    //
                    //  To avoid a breaking change, the support of localized component names (without explicit
                    //  Uppercase classification) is behind a `SupportLocalizedComponentNames` feature flag.
```